api/v1alpha1/networkfence_types.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FencingOperationResult string
+
+const (
+	// FencingOperationResultSucceeded represents the Succeeded operation state.
+	FencingOperationResultSucceeded FencingOperationResult = "Succeeded"
+
+	// FencingOperationResultFailed represents the Failed operation state.
+	FencingOperationResultFailed FencingOperationResult = "Failed"
+)
+
+// SecretSpec defines the secrets to be used for the network fencing operation.
+type SecretSpec struct {
+	// Name specifies the name of the secret.
+	Name string `json:"name,omitempty"`
+
+	// Namespace specifies the namespace in which the secret
+	// is located.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// NetworkFenceSpec defines the desired state of NetworkFence
+type NetworkFenceSpec struct {
+	// Driver contains  the name of CSI driver.
+	// +kubebuilder:validation:Required
+	Driver string `json:"driver"`
+
+	// Cidr contains a list of CIDR blocks, which are required to be fenced.
+	// +kubebuilder:validation:Required
+	Cidr []string `json:"cidr"`
```

Comment:
now that this is a list :partying_face: the name needs to be `cidrs` (with `s`)

Reply:
Thanks! Updated now